crates/compute/src/memory.rs

```diff
@@ -89,18 +95,20 @@ pub trait ComputeMemory<F> {
 	/// This allows the immutable reference to be copied.
 	fn as_const<'a>(data: &'a Self::FSliceMut<'_>) -> Self::FSlice<'a>;
 
-	/// Borrows a subslice of an immutable memory slice.
+	/// Borrows a subslice of an immutable memory slice
 	///
 	/// ## Preconditions
 	///
-	/// - the range bounds must be multiples of [`Self::MIN_SLICE_LEN`]
+	/// - the resulting slice must be either aligned by [`Self::ALIGNMENT`] or have a length
```

Comment:
This precondition is weird. I'd rather take the approach where `ALIGNMENT` is required to be power-of-two and we have `split_half` & `split_half_mut`.

Reply:
That would simplify the implementation a bit, but as an interface that methods looks like an ad-hoc solution - a special splitting logic for the special case. Moreover I'm not sure if it covers all the cases, probably when splitting into equal chunks we may all face the case when a single chunks is smaller than packing width.

To me the interface where all the splitting functions follow the same rules looks more consistent. What do you think?

Reply:
I'm pretty confident it will cover all of our needs. The buffers we operate on are all power-of-two sized, and when we split them, we divide into a power-of-two number of sub-blocks. The main exception is the allocator, which could work around this if a small buffer is ever requested.